### PR TITLE
Document publishing process

### PR DESCRIPTION
**Pull request by sstone**
This PR documents our process for  publishing artifacts to maven central. This process is manual for now: you must

- download artifacts generated by github CI
- sign them locally if needed
- upload them to a sonatype repository
- (release only) release and close the uploaded artifatcs from sonatype

I did not want to automate this process because it would mean adding secrets (sonatype credentials, gpg keys) to our github repositories.

### CODE REVIEW

File: publishing/PUBLISHING.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+To publish snapshot, you must add your sonatype credentials for the `ossrh` server to your local maven settings (typically in $HOME/.m2/settings.xml)
+
+- Download `snapshot.zip` generated by the `Publish snapshot` github action
+- unzip `snapshot.zip` in the `publishing` directory
+- edit `secp256k1-kmp-snapshot-deploy.sh` and update the `VERSION` environment variable if needed
+- run `secp256k1-kmp-snapshot-deploy.sh`
+
+## releases
+
+Releases are published to the Sonatype staging repository. If all items are valid they will the be published to `mmaven central` repository.
```
Comment (t-bast) [inline publishing/PUBLISHING.md:15]: nit:
```suggestion
Releases are published to the Sonatype staging repository. If all items are valid they will be published to `maven central` repository.
```

File: publishing/PUBLISHING.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+You must edit `secp256k1-kmp-staging-upload.sh` and add your sonatype credentials. You must also have a valid GPG key.
+
+- Download `release.zip` generated by the `Publish release` github action (which is triggered every time you publish a github release)
+- unzip `release.zip` in the `publishing` directory
+- edit `secp256k1-kmp-staging-upload.sh` and update the `VERSION` environment variable if needed
+- sign all artifacts with a valid gpg key: `find release -type f -print -exec gpg -ab {} \;`
+- run `secp256k1-kmp-staging-upload.sh`
+- log into sonatype, close and publish your staging repository. Artifacts will be available on Maven Central within a few hours.
```
Comment (t-bast) [inline publishing/PUBLISHING.md:23]: nit: do you mean `clone`?
```suggestion
- log into sonatype, clone and publish your staging repository. Artifacts will be available on Maven Central within a few hours.
```

**Review by t-bast — COMMENTED** [review]
I haven't actually tried publishing, but at least the instructions are there, if someday someone needs to publish and something fails, they will make a PR to fix what's broken :+1:

File: publishing/PUBLISHING.md (hunk 2 above)
Comment (sstone) [reply publishing/PUBLISHING.md:23]: It is sonatype's process: upload artifacts to a temporary "staging" repository, verify the contents of this staging repo, and if everything is ok then "close" and "release" it which will publish its content to maven central.

**Review by t-bast — APPROVED** [review]

**Pull request merged**